src/libstore/unix/build/local-derivation-goal.cc

```diff
@@ -814,7 +802,59 @@ static void handleChildException(bool sendException)
     }
 }
 
-void LocalDerivationGoal::startBuilder()
+static std::function<bool(Descriptor, std::string_view)> build_handler(
+    std::shared_ptr<BufferedSink> & logSink,
+    Activity & act,
+    std::map<ActivityId, Activity> & builderActivities,
+    bool & failed,
+    Descriptor & builderOut,
+    unsigned long & logSize,
+    std::string & currentLogLine,
+    size_t & currentLogLinePos,
+    std::list<std::string> & logTail
```

Comment:
This is a lot of function arguments. Maybe this could be moved into a struct? (Possibly in a future PR.)

Reply:
@edolstra I think the end game is just to move the closure inline, and all of this stuff can just become implicit captures.